### DISCUSSION TITLE
Proper way to respect JAVA_HOME if set

### DISCUSSION THREAD

**Discussion by mcred** [Homebrew/discussions#3061] [Writing Formulae/Casks] [answered]
For formulae, what is the proper way to respect the existing `JAVA_HOME` environment variable if it is set? When looking at different formulae, they do many different things for `JAVA_HOME`. `JAVA_HOME: Formula["openjdk"].opt_prefix` is pretty straight forward and doesn't allow for a user to set their own `JAVA_HOME` value. Does `JAVA_HOME:  "${JAVA_HOME:-#{Formula["openjdk"].opt_prefix}}"` do that?

**Comment by SMillerDev** [comment, marked as answer]
That's the idea, but we also have `overridable_java_home_env` for that

**Comment by mcred** [comment]
Are `Language::Java.overridable_java_home_env` and `JAVA_HOME: "${JAVA_HOME:-#{Formula["openjdk"].opt_prefix}}"` equivalent? Which is the preferred way to a default `JAVA_HOME` but take the system env as the priority?

**Reply by SMillerDev** [reply]
They are equivalent. As for the preferred way, I think the homebrew function is more readable. But that's just personal preference.

**Reply by mcred** [reply]
Thank you!
